pisa/stages/abs/pi_earth_absorption.py

```diff
@@ -0,0 +1,244 @@
+"""
+PISA pi stage for the calculation of earth layers and survival probabilities.
+
+The stage calculates first the depth of a water column that is mass-equivalent
+to the path traversed by the neutrino through the earth. This is done
+using the same Layers module that is also used for oscillation.
+The survival probability is then calculated from the average cross-section
+with protons and neutrons.
+"""
+from __future__ import absolute_import, print_function, division
+
+import numpy as np
+import ROOT
+
+from numba import guvectorize
+
+from pisa import FTYPE, TARGET
+from pisa.core.pi_stage import PiStage
+from pisa.utils.log import logging
+from pisa.utils.profiler import profile
+from pisa.stages.osc.layers import Layers
+from pisa.utils.numba_tools import WHERE
+from pisa.utils import vectorizer
+from pisa.utils.resources import find_resource
+from pisa.core.map import Map
+
+class pi_earth_absorption(PiStage):
+    """
+    earth absorption PISA Pi class
+
+    Paramaters
+    ----------
+    earth_model : PREM file path
+    xsec_file : path to ROOT file containing cross-sections
+    detector_depth : detector depth in km
+    prop_height : height of neutrino production in the atmosphere in km
+    
+    Notes
+    -----
+    
+    """
+    def __init__(self,
+                 data=None,
+                 params=None,
+                 input_names=None,
+                 output_names=None,
+                 debug_mode=None,
+                 input_specs=None,
+                 calc_specs=None,
+                 output_specs=None,
+                ):
+
+        expected_params = ('earth_model',
+                           'xsec_file',
+                           'detector_depth',
+                           'prop_height',
+                          )
+
+        input_names = ()
+        output_names = ()
+
+        # what are the keys used from the inputs during apply
+        input_apply_keys = ('weights', )
+        # what are keys added or altered in the calculation used during apply
+        # The weights are simply scaled by the earth survival probability
+        output_calc_keys = ('survival_prob', )
+        # what keys are added or altered for the outputs during apply
+        output_apply_keys = ('weights', )
+
+        # init base class
+        super(pi_earth_absorption, self).__init__(data=data,
+                                                  params=params,
+                                                  expected_params=expected_params,
```

Comment:
seems like there is some formatting gone wrong here (?)

Reply:
I'm not sure what you mean, but I changed the formatting now. It it better?